src/ex1/script.js

```diff
@@ -0,0 +1,55 @@
+const input = document.querySelector(".todo-input");
+const addButton = document.querySelector(".todo-button");
+const todoList = document.querySelector(".todo-list");
+
+addButton.addEventListener("click", () => addTodo(input));
+input.addEventListener("keypress", (e) => {
+  if (e.key === "Enter") {
+    addTodo(input);
+  }
+});
+
+function addTodo(input) {
+  if (input.value.length === 0) {
+    changeInputPlaceholder(false);
+    return;
+  }
+  changeInputPlaceholder(true);
+  const todoItem = document.createElement("div");
+  const newTodo = document.createElement("li");
+  const deleteButton = document.createElement("button");
+  const trashIcon = document.createElement("i");
+
+  newTodo.taskName = input.value;
+  newTodo.innerHTML = newTodo.taskName.substring(0, 20);
+  if (newTodo.taskName.length > 20) {
+    newTodo.innerHTML += "...";
+  }
```

Comment:
a suggestion - you can also do this with css and then it will be more correct, according to the actual text size -

max-width: 10em;
overflow: hidden;
white-space: nowrap;
text-overflow: ellipsis;

Reply:
text-overflow wont affect to todo-item class.